Tests/SwiftCurrent_SwiftUITests/SwiftCurrent_SwiftUITests.swift

```diff
@@ -324,154 +364,118 @@ final class SwiftCurrent_SwiftUIConsumerTests: XCTestCase {
         }
         let expectOnFinish = expectation(description: "OnFinish called")
+        let expectedArgs = UUID().uuidString
         let expectViewLoaded = ViewHosting.loadView(
-            WorkflowView(isLaunched: .constant(true))
+            WorkflowView(isLaunched: .constant(true), startingArgs: expectedArgs)
                 .thenProceed(with: WorkflowItem(FR1.self))
                 .thenProceed(with: WorkflowItem(FR2.self))
                 .onFinish { _ in
             expectOnFinish.fulfill()
         }).inspection.inspect { viewUnderTest in
-            XCTAssertNoThrow(try viewUnderTest.find(FR1.self).actualView().proceedInWorkflow())
-            XCTAssertNoThrow(try viewUnderTest.vStack().callOnAppear())
+            XCTAssertEqual(try viewUnderTest.find(FR1.self).text().string(), "FR1 type")
+            XCTAssertNoThrow(try viewUnderTest.find(FR1.self).actualView().proceedInWorkflow(.args(expectedArgs)))
+            XCTAssertEqual(try viewUnderTest.find(FR2.self).text().string(), "FR2 type")
             XCTAssertNoThrow(try viewUnderTest.find(FR2.self).actualView().proceedInWorkflow())
         }
 
         wait(for: [expectOnFinish, expectViewLoaded], timeout: 0.3)
     }
-    
-    func testWorkflowRelaunchesWhenSubsequentlyLaunched() throws {
-        struct FR1: View, FlowRepresentable, Inspectable {
-            var _workflowPointer: AnyFlowRepresentable?
-            var body: some View { Text("FR1 type") }
-        }
-        struct FR2: View, FlowRepresentable, Inspectable {
-            var _workflowPointer: AnyFlowRepresentable?
-            var body: some View { Text("FR2 type") }
-        }
-        let binding = Binding(wrappedValue: true)
-        let expectOnFinish = expectation(description: "OnFinish called")
-        let expectViewLoaded = ViewHosting.loadView(
-            WorkflowView(isLaunched: binding)
-                .thenProceed(with: WorkflowItem(FR1.self))
-                .thenProceed(with: WorkflowItem(FR2.self))
-                .onFinish { _ in
-                    expectOnFinish.fulfill()
-                }).inspection.inspect { viewUnderTest in
-                    XCTAssertNoThrow(try viewUnderTest.find(FR1.self).actualView().proceedInWorkflow())
-                    let vStack = try viewUnderTest.vStack()
-                    binding.wrappedValue = false
-                    XCTAssertNoThrow(try vStack.callOnDisappear())
-                    binding.wrappedValue = true
-                    XCTAssertNoThrow(try viewUnderTest.vStack().callOnAppear())
-                    XCTAssertNoThrow(try viewUnderTest.find(FR1.self).actualView().proceedInWorkflow())
-                    XCTAssertNoThrow(try viewUnderTest.find(FR2.self).actualView().proceedInWorkflow())
-                }
-        
-        wait(for: [expectOnFinish, expectViewLoaded], timeout: 0.3)
-    }
 
-    func testWorkflowMaintainsStateWhenViewDisappearsAndReappears_WithoutIsLaunchedChanging() throws {
```

Comment:
Git might be making it hard to track where this test went, but this is still a valid-ish test (probably not the way it was being tested).  It's important that if our views roll off the screen and then come back, they haven't reset their state.

Reply:
Correct, and that's a function of SwiftUI at this point. Because we do not have an `onAppear` or `onDisappear` modifier that change anything all behavior will be as Apple has intended. BOW DOWN TO THEIR IDEAS

Reply:
I'll bow down to their ideas.  In doing so, I'd like to confirm that if WorkflowView is placed in a lazyVStack and scrolled off the screen and then comes back into view, the user is still in the same spot of their workflow.  A test would be amazing, but also just confirming that has been tested would be enough in this situation.

Reply:
I'm unaware of how to do such a test without pulling in XCUITest and a hosting application. If that flakified test is worthwhile to you we can do it, elsewise it might be just a matter of pulling in the example app and proving it there.

Reply:
Ok.  Let's leave this open until someone validates it.  I think we'll be good enough with just a solid check of this (100 items or more would be preferable).  No flakey test needed.

Whoever validates this please respond here and resolve the conversation.

Reply:
Confirmed that this works exactly as expected with a LazyVStack and looooots of scrolling.